File: src/main/scala/ru/tinkoff/gatling/amqp/request/AmqpDslBuilderBase.scala
package ru.tinkoff.gatling.amqp.request

import io.gatling.core.config.GatlingConfiguration
import io.gatling.core.session._

case class AmqpDslBuilderBase(requestName: Expression[String]) {
  def publish(implicit configuration: GatlingConfiguration)      = PublishDslBuilderExchange(requestName, configuration)
  def requestReply(implicit configuration: GatlingConfiguration) =
    RequestReplyDslBuilderExchange(requestName, configuration)
}

File: src/main/java/ru/tinkoff/gatling/javaapi/request/AmqpDslBuilderBase.java
package ru.tinkoff.gatling.javaapi.request;

public final class AmqpDslBuilderBase {
    private final ru.tinkoff.gatling.amqp.request.AmqpDslBuilderBase wrapped;

    public AmqpDslBuilderBase(ru.tinkoff.gatling.amqp.request.AmqpDslBuilderBase wrapped){
        this.wrapped = wrapped;
    }

    public PublishDslBuilderExchange publish(){
        return new PublishDslBuilderExchange(wrapped.publish(io.gatling.core.Predef.configuration()));
    }
    public RequestReplyDslBuilderExchange requestReply(){
        return new RequestReplyDslBuilderExchange(wrapped.requestReply(io.gatling.core.Predef.configuration()));
    }
}
